### PR TITLE
Add message prefixes

### PR DESCRIPTION
**Pull request by Hyodar**
This PR adds prefixes to messages and task responses with the objective of making it even more difficult to have collisions with other messaging formats. Not all AVSs do this, but it's a small cost for a huge benefit. Now, these hashes are `hash(prefix, hash(data))`, which also covers the possibility of a similar data format including a 32-byte field.
I'm following a simpler approach than EIP712 because in some cases we wouldn't benefit too much from the features, and in some cases, such as in `verifyingContract`, we plainly don't want them so we don't have a digest for every chain.

### CODE REVIEW

File: core/utils.go [hunk 1]
```diff
@@ -19,6 +19,32 @@ func Keccak256(data []byte) ([32]byte, error) {
 	return digest, nil
 }
 
+func HashWithPrefix(prefix []byte, data []byte) ([32]byte, error) {
+	prefixHash, err := Keccak256(prefix)
```
Comment (taco-paco) [inline core/utils.go:23]: Keccak shalln't return an error.

Comment (taco-paco) [reply core/utils.go:23]: Implementation also doesn't

**Review by taco-paco — APPROVED** [review]

File: core/utils.go (hunk 1 above)
Comment (Hyodar) [reply core/utils.go:23]: Indeed - checking this out

Comment (Hyodar) [reply core/utils.go]: Done in [e9c4928](https://github.com/NethermindEth/near-sffl/pull/104/commits/e9c4928ff7f1c5b675af7c592a98376a0f50d085)

**Pull request merged**